File: app/src/main/java/oauth/signpost/commonshttp/HttpResponseAdapter.java
package oauth.signpost.commonshttp;

import java.io.IOException;
import java.io.InputStream;


public class HttpResponseAdapter implements oauth.signpost.http.HttpResponse {

    private cz.msebera.android.httpclient.HttpResponse response;

    public HttpResponseAdapter(cz.msebera.android.httpclient.HttpResponse response) {
        this.response = response;
    }

    public InputStream getContent() throws IOException {
        return response.getEntity().getContent();
    }

    public int getStatusCode() throws IOException {
        return response.getStatusLine().getStatusCode();
    }

    public String getReasonPhrase() throws Exception {
        return response.getStatusLine().getReasonPhrase();
    }

    public Object unwrap() {
        return response;
    }
}

File: app/src/main/java/oauth/signpost/commonshttp/HttpResponseAdapter.kt
package oauth.signpost.commonshttp

import cz.msebera.android.httpclient.HttpResponse
import java.io.IOException
import java.io.InputStream

class HttpResponseAdapter(private val response: HttpResponse) : oauth.signpost.http.HttpResponse {
    @Throws(IOException::class)
    override fun getContent(): InputStream? {
        return response.entity.content
    }

    @Throws(IOException::class)
    override fun getStatusCode(): Int {
        return response.statusLine.statusCode
    }

    @Throws(Exception::class)
    override fun getReasonPhrase(): String? {
        return response.statusLine.reasonPhrase
    }

    override fun unwrap(): Any {
        return response
    }
}